runtime/src/main/java/io/quarkiverse/quinoa/QuinoaDevProxyHandler.java

```diff
@@ -58,8 +67,73 @@ public void handle(final RoutingContext ctx) {
             next(currentClassLoader, ctx);
             return;
         }
-        final String uri = computeURI(resourcePath, request);
+
         final MultiMap headers = request.headers();
+        final String uri = computeURI(resourcePath, request);
+
+        // WebSocket connection for Hot Module Reloading (HMR)
+        if (!config.prodMode && headers.get("Connection").contains("Upgrade")) {
+            // Vite uses the root URL "/" which is special
+            final String forwardUri = request.uri().equals("/") ? "/" : uri;
+            // some servers use sub-protocols like Vite which must be forwarded
+            final String subProtocol = headers.get("Sec-WebSocket-Protocol");
+            final List<String> subProtocols = new ArrayList<>(1);
+            MultiMap socketHeaders;
+            if (StringUtil.isNullOrEmpty(subProtocol)) {
+                socketHeaders = null;
+            } else {
+                socketHeaders = headers;
+                subProtocols.add(subProtocol);
+                LOG.infof("Sec-WebSocket-Protocol: %s", subProtocol);
+            }
+            LOG.infof("Quinoa is forwarding web socket: '%s'", forwardUri);
+            final String host = request.localAddress().host();
+            final Future<ServerWebSocket> fut = request.toWebSocket();
+            fut.onSuccess(serverWs -> {
+                final WebSocketConnectOptions options = new WebSocketConnectOptions()
+                        .setHost(host)
+                        .setPort(port)
+                        .setURI(forwardUri)
+                        .setHeaders(socketHeaders)
+                        .setSubProtocols(subProtocols)
+                        .setAllowOriginHeader(false);
+
+                LOG.infof("Quinoa client websocket: %s:%s%s", host, port, forwardUri);
+                httpClient.webSocket(options, clientContext -> {
+                    if (clientContext.succeeded()) {
+                        try {
+                            final WebSocket clientWs = clientContext.result();
```

Comment:
What if it's a websocket connection made to be handled by the Quarkus backend? (and not a HMR)

Reply:
You can still use your `ignorePaths` value to ignore any WebSocket connection you don't want touched.   But I have a feeling this code is going to work no matter what.